x-pack/platform/plugins/shared/onechat/public/application/components/tools/execute/test_tools.tsx

```diff
@@ -0,0 +1,217 @@
+                    key={field.name}
+                    label={field.label}
+                    isInvalid={!!errors[field.name]}
+                    error={errors[field.name]?.message as string}
+                  >
+                    {field.type === 'integer' ||
+                    field.type === 'long' ||
+                    field.type === 'double' ||
+                    field.type === 'float' ? (
+                      <Controller
+                        control={form.control}
+                        name={field.name}
+                        rules={{ required: `${field.label} is required` }}
+                        render={({ field: { ref, ...fieldProps }, fieldState: { error } }) => (
+                          <EuiFieldNumber
+                            {...fieldProps}
+                            inputRef={ref}
+                            placeholder={`Enter ${field.label.toLowerCase()}`}
+                            fullWidth
+                            isInvalid={!!error}
+                          />
+                        )}
+                      />
+                    ) : (
+                      <Controller
+                        control={form.control}
+                        name={field.name}
+                        rules={{ required: `${field.label} is required` }}
+                        render={({ field: { ref, ...fieldProps }, fieldState: { error } }) => (
+                          <EuiFieldText
+                            {...fieldProps}
+                            inputRef={ref}
+                            placeholder={`Enter ${field.label.toLowerCase()}`}
+                            fullWidth
+                            isInvalid={!!error}
+                          />
+                        )}
+                      />
+                    )}
+                  </EuiFormRow>
+                ))}
+                <EuiSpacer size="m" />
+                <EuiButton
+                  type="submit"
+                  size="s"
+                  fill
+                  isLoading={isExecuting}
+                  disabled={Object.keys(errors).length > 0 || !tool?.id}
+                >
+                  {i18n.translate('xpack.onechat.tools.testTool.executeButton', {
+                    defaultMessage: 'Submit',
+                  })}
+                </EuiButton>
+              </EuiForm>
+            </EuiFlexItem>
+            <EuiFlexItem>
+              <div style={{ marginLeft: '30px' }}>
+                <EuiTitle size="s">
+                  <h5>
+                    {i18n.translate('xpack.onechat.tools.testTool.responseTitle', {
+                      defaultMessage: 'Response',
+                    })}
+                  </h5>
+                </EuiTitle>
+              </div>
+              <EuiSpacer size="m" />
+              <CodeEditor
+                languageId="json"
+                value={response}
+                fullWidth={true}
+                height="calc(100vh - 275px)"
```

Comment:
This part feels a bit tricky, manually calculating CSS height seems risky.

From what I understand, the code editor requires an explicit height. Why not set it to 100%? Could we instead use flexbox to let it stretch automatically?

Also, since the result is read-only, is there a reason we’re using the `CodeEditor` component (which is meant for dynamic input)? Wouldn’t a static code block be simpler and more appropriate here?